pkg/sdk/roles_validations.go

```diff
@@ -0,0 +1,93 @@
+package sdk
+
+import "errors"
+
+func (opts *CreateRoleOptions) validate() error {
```

Comment:
please add 
```
var (
	_ validatable = new(CreateRoleOptions)
	...
)
```
block (we have discussed and agreed to have this compiler check present)